kms-message/src/kms_message_private.h

```diff
@@ -45,13 +46,37 @@ struct _kms_request_t {
    /* turn off for tests only, not in public kms_request_opt_t API */
    bool auto_content_length;
    _kms_crypto_t crypto;
+   kms_request_str_t *to_string;
    kms_request_provider_t provider;
+
+   /* TODO (MONGOCRYPT-342): make a union for each KMS provider type.
+      kms_request_provider_t provider;
+      union {
+         struct {} aws;
+         struct {} azure;
+         struct {} gcp;
+         struct {} kmip;
+      }
+   */
+   struct {
+      uint8_t *data;
+      uint32_t len;
+   } kmip;
 };
 
 struct _kms_response_t {
    int status;
    kms_kv_list_t *headers;
    kms_request_str_t *body;
+
+   /* TODO (MONGOCRYPT-???): make a union for each KMS provider type. */
```

Comment:
Similar to MONGOCRYPT-342 (make `kms_request_t` a union), I think `kms_response_t` could use a union to store KMS provider specific data, determined by the `provider` field. I plan to file a ticket to address this technical debt in follow-up.

Reply:
Filed MONGOCRYPT-347.